testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/EnvironmentProvider.java

```diff
@@ -53,17 +56,30 @@ protected EnvironmentProvider(List<EnvironmentExtender> bases)
                 .build();
 
         Set<EnvironmentExtender> seen = newSetFromMap(new IdentityHashMap<>());
-        extenders.forEach(extender -> extend(extender, builder, seen));
+        extenders.forEach(extender -> extend(extender, builder, extraOptions, seen));
         return builder;
     }
 
-    private void extend(EnvironmentExtender extender, Environment.Builder builder, Set<EnvironmentExtender> seen)
+    private void extend(EnvironmentExtender extender, Environment.Builder builder, Map<String, String> extraOptions, Set<EnvironmentExtender> seen)
     {
         extender.getDependencies()
-                .forEach(dependencyExtender -> extend(dependencyExtender, builder, seen));
+                .forEach(dependencyExtender -> extend(dependencyExtender, builder, extraOptions, seen));
         if (seen.add(extender)) {
-            log.info("Building environment %s with extender: %s", builder.getEnvironmentName(), extender.getClass().getSimpleName());
-            extender.extendEnvironment(builder);
+            Optional<Class<?>> configClass = extender.getConfigClass();
```

Comment:
Maybe instead we could have something like `extender.setExtraOptions(Map<String, String>)` which throws on default

Reply:
We could drop `extendEnvironment` overload and `getConfigClass` then and left "decoding" logic to the extender itself.

Reply:
That should also work.

The only reason I added the config class was to provide some way for the extender to let callers know that it will accept options. But with your `setExtraOptions` idea that will not be required anymore.